devicetypes/smartthings/zigbee-water-leak.src/zigbee-water-leak.groovy

```diff
@@ -0,0 +1,149 @@
+
+		for (int i = 0; i <= 90; i += 10) {
+			status "battery 0021 0x${i}": "read attr - raw: 8C900100010A21000020C8, dni: 8C90, endpoint: 01, cluster: 0001, size: 0A, attrId: 0021, result: success, encoding: 20, value: ${i}"
+		}
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"water", type: "generic", width: 6, height: 4){
+			tileAttribute ("device.water", key: "PRIMARY_CONTROL") {
+				attributeState "dry", icon:"st.alarm.water.dry", backgroundColor:"#ffffff"
+				attributeState "wet", icon:"st.alarm.water.wet", backgroundColor:"#00a0dc"
+			}
+		}
+
+		valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+			state "battery", label:'${currentValue}% battery', unit:""
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main "water"
+		details(["water", "battery", "refresh"])
+	}
+}
+
+def parse(String description) {
+	log.debug "description: $description"
+
+	def result
+	Map map = zigbee.getEvent(description)
+
+	if (!map) {
+		if (description?.startsWith('zone status')) {
+			map = getMoistureResult(description)
+		} else if(description?.startsWith('enroll request')){
+			List cmds = zigbee.enrollResponse()
+			log.debug "enroll response: ${cmds}"
+			result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+		}else {
+			Map descMap = zigbee.parseDescriptionAsMap(description)
+            log.debug "${descMap}"
+			if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002) {
+				map = getMoistureResult(description)
+			} else if (descMap?.clusterInt == 0x0001 && descMap?.attrInt == 0x0021 && descMap?.commandInt != 0x07 && descMap?.value) {
+				map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+			}
+		}
+	}
+	if(map&&!result){
+		result = createEvent(map)
+	}
+	log.debug "Parse returned $result"
+
+	result
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	log.debug "Refreshing Values"
+	def refreshCmds = []
+	refreshCmds += zigbee.enrollResponse()
+
+	refreshCmds
+}
+
+def installed(){
+	log.debug "call installed()"
+	sendEvent(name: "checkInterval", value: 20 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+}
+
+def configure() {
+	sendEvent(name: "checkInterval", value: 20 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.debug "Configuring Reporting"
+	                
+	return refresh() 
+}
+
+def getMoistureResult(description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+	def value = zs?.isAlarm1Set()?"wet":"dry"
+	[
+		name           : 'water',
+		value          : value,
+		descriptionText: "${device.displayName} is $value",
+		translatable   : true
+	]
+}
+
+def getBatteryPercentageResult(rawValue) {
```

Comment:
Does this device support the battery percentage remaining attribute? The fingerprint doesn't include cluster 0x0001 which is the cluster that contains that attribute.